src/relayserver/RelayServer.ts

```diff
@@ -607,14 +612,10 @@ export class RelayServer extends EventEmitter {
       debug(`can't register yet: auth=${this.authorizedHub} stake=${this.stake.toString()}`)
       return receipts
     }
-    const workersAddedEvents = await this.relayHubContract?.contract.getPastEvents('RelayWorkersAdded', {
-      fromBlock: 1,
-      filter: { relayManager: this.managerAddress }
-    })
 
     // add worker only if not already added
-    if (workersAddedEvents.find((e: any) => e.returnValues.newRelayWorkers
-      .map((a: string) => a.toLowerCase()).includes(this.getAddress(1).toLowerCase())) == null) {
+    const workersAdded = await this._areWorkersAdded()
```

Comment:
what's going on here? why _isRegistered and _areWorkersAdded need to be async?
is there a risk of checking and mining race condition then?

Reply:
I just separated it to a function, they are async since `getPastEvents()` is called